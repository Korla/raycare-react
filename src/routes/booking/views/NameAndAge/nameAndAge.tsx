import * as React from 'react';
import { Panel } from '../../../../common';

export interface NameAndAgeProps {
  name: string;
  age: number;
}

export interface NameAndAgeDispatch {
  addAge: () => void;
}

export class NameAndAge extends React.Component<NameAndAgeProps & NameAndAgeDispatch, void> {
  render() {
    return (
      <Panel title="Name and age">
        <div>
          <span>
            Name: {this.props.name}, Age: {this.props.age}
          </span>
          <button onClick={this.props.addAge}>Add</button>
        </div>
      </Panel>
    );
  }
}
