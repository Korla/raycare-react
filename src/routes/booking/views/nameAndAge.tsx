import * as React from 'react';
import { connect, Dispatch } from 'react-redux';
import { BookingState } from '../index';
import { addAge } from '../actions';

interface NameAndAgeProps {
  name: string;
  age: number;
}

interface NameAndAgeDispatch {
  addAge: () => void;
}

class NameAndAge extends React.Component<NameAndAgeProps & NameAndAgeDispatch, void> {
  render() {
    return (
      <div>
        Name: {this.props.name}, Age: {this.props.age}
        <button onClick={this.props.addAge}>Add</button>
      </div>
    );
  }
}

const mapStateToProps = ({ name, age }: BookingState) => ({ name, age });

const mapDispatchToProps = (dispatch: Dispatch<void>) => {
  return {
    addAge: () => {
      dispatch(addAge());
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(NameAndAge);
