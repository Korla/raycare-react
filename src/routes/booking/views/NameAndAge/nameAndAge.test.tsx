import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { NameAndAge } from './nameAndAge';

test('nameAndAge renders the name and age', () => {
  // Arrange
  const { name, age, jsx } = setup();
  const renderedNameAndAge = shallow(jsx);

  // Act
  const nameAndAgeText = renderedNameAndAge.find('span').text();

  // Act and assert
  expect(nameAndAgeText).toEqual(`Name: ${name}, Age: ${age}`);
});

test('nameAndAge snapshot test', () => {
  // Arrange
  const { jsx } = setup();

  // Act
  const tree = renderer.create(jsx).toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});

function setup() {
  const name = 'A name';
  const age = 5;
  const addAge = () => undefined;

  const jsx = <NameAndAge name={name} age={age} addAge={addAge} />;

  return { name, age, addAge, jsx };
}