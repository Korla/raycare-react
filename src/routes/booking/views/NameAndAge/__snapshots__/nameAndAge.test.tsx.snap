exports[`test nameAndAge snapshot test 1`] = `
<div
  style={
    Object {
      "background": "#777",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <div
    className="panel-header"
    style={
      Object {
        "borderBottom": "1px solid white",
        "padding": 10,
      }
    }>
    <span
      style={
        Object {
          "cursor": "pointer",
          "textDecoration": "underline",
        }
      }>
      Name and age
    </span>
  </div>
  <div
    style={
      Object {
        "padding": 10,
      }
    }>
    <div>
      <span>
        Name: 
        A name
        , Age: 
        5
      </span>
      <button
        onClick={[Function]}>
        Add
      </button>
    </div>
  </div>
</div>
`;
